artifacts/writer.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+"""The artifact writer objects."""
+
+import abc
+import json
+import yaml
+
+class BaseArtifactsWriter(object):
+  """Class that implements the artifacts reader interface."""
+
+  @abc.abstractmethod
+  def _ConvertArtifact(self, artifact):
```

Comment:
Moving it to ArtifactDefinition as CopyToDict() is probably the better plan.  I will do that.